### ISSUE TITLE
UpdateUptimeTest calls don't seem to work

### ISSUE THREAD

**Issue by lachlanmunro** [StatusCakeDev/statuscake-go#20]
Thanks for the effort and making this available, much appreciated!

I've got an OAuth2 API that I need to monitor so I'm just slapping something together to keep the custom header token refreshed. I've been successfully using this library for the other tasks (checking the Uptime Test exists, getting a list of Uptime Tests etc) but have run into issues with the `UpdateUptimeTest` call. This is the first `application/x-www-form-urlencoded` call I've needed to make from your library.

Calling the `UpdateUptimeTest` library function seems to 500 and error (below) even when doing it by "hand" with the same configuration works correctly.

> 2022-10-25T20:43:33.113+01:00 FTL Failed to update Uptime Test authentication: failed to update uptime test: Unexpected error. Try again later. error="failed to update uptime test: Unexpected error. Try again later."

I've made the call successfully in order to confirm that the issue is not with the API. Below is a chunk of that code, the uncommented block is just calling the API using a `http.Client`, the lower commented out section is the library call to do the same thing (which I can't get to work, error above).

```golang
type StatusCakeClient struct {
	c   *statuscake.Client
	h   *http.Client
	cfg *config.StatusCake
}

// UpdateUptimeTestCustomHeader replaces the custom header of the supplied Uptime Test.
func (s *StatusCakeClient) UpdateUptimeTestCustomHeader(ctx context.Context, id, customHeader string) (err error) {
	requestBody := url.Values{}
	requestBody.Add("custom_header", customHeader)
	req, err := http.NewRequestWithContext(ctx, http.MethodPut, fmt.Sprintf("https://api.statuscake.com/v1/uptime/%v", id), strings.NewReader(requestBody.Encode()))
	if err != nil {
		return fmt.Errorf("failed to construct Uptime Test update request: %w", err)
	}

	req.Header.Set("content-type", "application/x-www-form-urlencoded")
	req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", s.cfg.Token))
	resp, err := s.h.Do(req)
	defer func() {
		if resp != nil && resp.Body != nil {
			if closeErr := resp.Body.Close(); closeErr != nil && err == nil {
				err = fmt.Errorf("failed to close response body: %w", closeErr)
			}
		}
	}()

	if err != nil {
		return fmt.Errorf("failed to send Uptime Test update request: %w", err)
	}

	if resp.StatusCode != 204 {
		bytes, err := io.ReadAll(resp.Body)
		if err != nil {
			return fmt.Errorf("%w (%v) and couldn't read body: %v", ErrNot20XResponse, resp.StatusCode, err)
		}

		return fmt.Errorf("%w (%v): %s", ErrNot20XResponse, resp.StatusCode, string(bytes))
	}

	return nil

	// Library call doesn't seem to work, 500s?
	//
	// r := s.c.UpdateUptimeTest(ctx, id)
	// r.CustomHeader(customHeader)
	// if err := r.Execute(); err != nil {
	// 	return fmt.Errorf("failed to update uptime test: %w", err)
	// }
	//
	// return nil
}
```

Am I calling this wrong from within the library?

**Comment by tomasbasham** [comment]
Are you able to provide an example of the `customHeader` field you're trying to update. The following code appeared to work for me.

```go
package main

import (
  "context"
  "fmt"

  "github.com/StatusCakeDev/statuscake-go"
  "github.com/StatusCakeDev/statuscake-go/credentials"
)

func main() {
  apiToken := "API TOKEN"

  bearer := credentials.NewBearerWithStaticToken(apiToken)
  client := statuscake.NewClient(statuscake.WithRequestCredentials(bearer))

  err := client.UpdateUptimeTest(context.Background(), "6664852").
    CustomHeader(`{"hello": "world"}`).
    Execute()
  if err != nil {
    printError(err)
  }
}

func printError(err error) {
  fmt.Println(err)
  fmt.Printf("%+v\n", statuscake.Errors(err))
}
```

**Comment by lachlanmunro** [comment]
Oh, I see it.

The `CustomHeader` call doesn't have a pointer receiver so it's not actually changing the calling "request" struct.

TLDR don't do:

```
r := s.c.UpdateUptimeTest(ctx, id)
r.CustomHeader(customHeader)
```

needs to be:

```
r := s.c.UpdateUptimeTest(ctx, id)
r = r.CustomHeader(customHeader)
```

or/etc :

```
r := s.c.UpdateUptimeTest(ctx, id).CustomHeader(customHeader)
```

Thanks for your help, your testcase teased it out. I'll close this out assuming this is by design (pros and cons).
